### PR TITLE
Paint surface

### PR DESCRIPTION
**Pull request by RobotCaleb**
(this is based on add_room branch)

This is mostly done.
Use to start, use again to finish
Alt-use to cycle replace mode
Long-use to cancel started

I didn't include removing surfaces with this. Feel free to hack that on if you want.

It's really quite cool and convenient.

### CODE REVIEW

File: src/tools/paint_surface.cc [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+        switch ((surface_index)index) {
+            case surface_xp:
+            case surface_xm:
+                if (start_block.x != bl.x && start_index != si)
+                    return false;
+                break;
+            case surface_yp:
+            case surface_ym:
+                if (start_block.y != bl.x && start_index != si)
```
Comment (chrisforbes) [inline src/tools/paint_surface.cc]: .y == .x ?

File: src/tools/paint_surface.cc [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+                    return false;
+                break;
+            case surface_zp:
+            case surface_zm:
+                if (start_block.z != bl.x && start_index != si)
```
Comment (chrisforbes) [inline src/tools/paint_surface.cc]: .z == .x ?

File: src/tools/paint_surface.cc [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+        i[1] = 1;
+    }
+
+    auto min = glm::min(start_block, end);
+    auto max = glm::max(start_block, end);
+
+    auto s = glm::value_ptr(min);
+    auto e = glm::value_ptr(max);
+    auto c = glm::value_ptr(cur);
+    for (auto j = s[i[0]]; j <= e[i[0]]; j++) {
```
Comment (chrisforbes) [inline src/tools/paint_surface.cc]: This is a bit nasty..

File: src/tools/paint_surface.cc (hunk 1 above)
Comment (RobotCaleb) [reply src/tools/paint_surface.cc]: Ah. Nice catch.

File: src/tools/paint_surface.cc (hunk 3 above)
Comment (RobotCaleb) [reply src/tools/paint_surface.cc]: Suggestions? I don't disagree but it seemed preferable to lots of checks.

**Pull request merged**